tests/acceptance/editor-test.js

```diff
@@ -233,33 +283,85 @@ describe('Acceptance: Editor', function() {
             await click(testSelector('publishmenu-save'));
+
+            expect(
+                find(testSelector('publishmenu-save')).text().trim(),
+                'publish menu save button updated after draft is scheduled'
+            ).to.equal('Scheduled');
+
             await click(testSelector('publishmenu-cancel'));
 
             expect(
                 find(testSelector('publishmenu-scheduled')),
                 'publish menu is not shown after closed'
             ).to.not.exist;
 
+            // expect countdown to show warning, that post will be published in x minutes
+            expect(find(testSelector('schedule-countdown')).text().trim(), 'notification countdown')
+                .to.contain('Post will be published in');
+
             expect(
                 find(testSelector('publishmenu-trigger')).text().trim(),
                 'scheduled publish button text'
             ).to.equal('Scheduled');
 
             expect(
                 find(testSelector('editor-post-status')).text().trim(),
-                'scheduled status text'
+                'scheduled post status'
             ).to.equal('Scheduled');
 
-            // expect countdown to show warning, that post will be published in x minutes
-            expect(find(testSelector('schedule-countdown')).text().trim(), 'notification countdown')
-                .to.contain('Post will be published in');
+            // Re-schedule
+            await click(testSelector('publishmenu-trigger'));
+            await click(testSelector('publishmenu-scheduled-option'));
+            expect(
+                find(testSelector('publishmenu-save')).text().trim(),
+                'scheduled post button reschedule text'
+            ).to.equal('Re-schedule');
+
+            await click(testSelector('publishmenu-save'));
+
+            expect(
+                find(testSelector('publishmenu-save')).text().trim(),
+                'publish menu save button text for a rescheduled post'
+            ).to.equal('Re-scheduled');
+
+            await click(testSelector('publishmenu-cancel'));
+
+            expect(
+                find(testSelector('publishmenu-scheduled')),
+                'publish menu is not shown after closed'
+            ).to.not.exist;
+
+            expect(
+                find(testSelector('editor-post-status')).text().trim(),
+                'scheduled status text'
+            ).to.equal('Scheduled');
 
             // unschedule
             await click(testSelector('publishmenu-trigger'));
             await click(testSelector('publishmenu-draft-option'));
+
+            expect(
+                find(testSelector('publishmenu-save')).text().trim(),
+                'publish menu save button updated after scheduled post is unscheduled'
+            ).to.equal('Un-schedule');
+
             await click(testSelector('publishmenu-save'));
+
+            // TODO Investigate- Practical testing works but test results in Re-scheduled text
```

Comment:
Does this TODO still need to be here? Passing test seems to contradict the TODO and I wasn't able to replicate bad behaviour in manual testing.